sqllineage/core/parser/sqlfluff/extractors/base.py

```diff
@@ -220,9 +222,17 @@ def extract_subquery(
                 if sq.query.get_child("with_compound_statement")
                 else SelectExtractor
             )
-            holder |= extractor_cls(self.dialect, self.metadata_provider).extract(
-                sq.query, AnalyzerContext(cte=holder.cte, write={sq})
-            )
+            subquery_holder = extractor_cls(
+                self.dialect, self.metadata_provider
+            ).extract(sq.query, AnalyzerContext(cte=holder.cte, write={sq}))
+
+            if (
+                SQLLineageConfig.LATERAL_COLUMN_ALIAS_REFERENCE
+                and bool(self.metadata_provider) is True
+            ):
+                subquery_holder.graph.add_node(sq, **{NodeTag.WRITE: False})
```

Comment:
This is open discussion. Is it possible that we move this logic to SubQueryLineageHolder `__or__` function?

I tried to do this (basically the same thing: to remove WRITE tag from SubQuery), but it breaks 6 test cases:
```python
    def __or__(self, other):
        self.graph = nx.compose(self.graph, other.graph)
        for t, attr in self.graph.nodes(data=True):
            if attr.get(NodeTag.READ, True) and attr.get(NodeTag.WRITE, True) and isinstance(t, SubQuery):
                self.graph.add_node(t, **{NodeTag.WRITE: False})
        return self
```

If somehow we can make it work, then we can start imagine in `select.py`, maybe we can do `self.extract_subquery(subqueries, holder)` ahead for all cases.

Reply:
May I ask what the reasons were for moving the code to that location?

I tried it, but it ended in failure because there was no inner subquery context in SubQueryLineageHolder `__or__` function, and the outer subquery was also introduced. so that it was impossible to uniquely identify the inner subquery that needed to remove the write flag.

Reply:
I just wanted to move `self.extract_subquery(subqueries, holder)` up universally in `select.py`. I see you already get it done in latest change. Good job. 

We don't need to move this part of logic then.